src/blob/index/simple.rs

```diff
@@ -79,10 +79,11 @@ where
         let file = File::create(path, ioring)
             .await
             .with_context(|| format!("file open failed {:?}", path))?;
-        file.write_append(&buf).await?;
+        file.write_append_all(buf.freeze()).await?;
         header.set_written(true);
-        let serialized_header = serialize(&header)?;
-        file.write_at(0, &serialized_header).await?;
+        let mut serialized_header = BytesMut::new();
```

Comment:
And here: It would be good to estimate expected size and create buffer with capacity. Bincode has `serialized_size` function for that